db/icat.go

```diff
@@ -54,70 +60,143 @@ SELECT id, root FROM child_mapping WHERE storage`
+	    WHEN coll_name LIKE '/' || $1 || '/trash/home/ipcservices/%%' THEN REGEXP_REPLACE(coll_name, '/' || $1 || '/trash/home/ipcservices/([^/]+).*', E'\\1')
+       END, coll_id
     FROM r_coll_main
-   WHERE coll_name LIKE '/' || $1 || '/home/' || $2 || '/%'
+   WHERE coll_name LIKE '/' || $1 || '/home/' || $2 || '/%%'
       OR coll_name =    '/' || $1 || '/home/' || $2
-      OR coll_name LIKE '/' || $1 || '/trash/home/' || $2 || '/%'
+      OR coll_name LIKE '/' || $1 || '/trash/home/' || $2 || '/%%'
       OR coll_name =    '/' || $1 || '/trash/home/' || $2
-      OR coll_name LIKE '/' || $1 || '/trash/home/de-irods/' || $2 || '/%'
+      OR coll_name LIKE '/' || $1 || '/trash/home/de-irods/' || $2 || '/%%'
       OR coll_name =    '/' || $1 || '/trash/home/de-irods/' || $2
-      OR coll_name LIKE '/' || $1 || '/trash/home/ipcservices/' || $2 || '/%'
+      OR coll_name LIKE '/' || $1 || '/trash/home/ipcservices/' || $2 || '/%%'
       OR coll_name =    '/' || $1 || '/trash/home/ipcservices/' || $2
-`
+`, table)
 
-	log.Tracef("createSpecificUserColls SQL: %s, [%s %s]", q, i.zone, u)
+	log.Tracef("populateSpecificUserColls SQL: %s, [%s %s]", q, i.zone, username)
 
-	_, err := i.db.ExecContext(context, q, i.zone, u)
+	_, err := i.db.ExecContext(context, q, i.zone, username)
 	if err != nil {
-		return errors.Wrap(err, "Error creating user_colls table for user")
+		return errors.Wrap(err, "Error filling user_colls table for user")
 	}
+	return nil
+}
 
+func (i *ICATDatabase) populateBatchUserColls(context context.Context, start, end, table string) error {
+	q := fmt.Sprintf(`INSERT INTO %s (user_name, coll_id)
+SELECT CASE WHEN coll_name LIKE '/' || $1 || '/home/%%' THEN REGEXP_REPLACE(coll_name, '/' || $1 || '/home/([^/]+).*', E'\\1')
+            WHEN coll_name LIKE '/' || $1 || '/trash/home/%%' THEN REGEXP_REPLACE(coll_name, '/' || $1 || '/trash/home/([^/]+).*', E'\\1')
+	    WHEN coll_name LIKE '/' || $1 || '/trash/home/de-irods/%%' THEN REGEXP_REPLACE(coll_name, '/' || $1 || '/trash/home/de-irods/([^/]+).*', E'\\1')
+	    WHEN coll_name LIKE '/' || $1 || '/trash/home/ipcservices/%%' THEN REGEXP_REPLACE(coll_name, '/' || $1 || '/trash/home/ipcservices/([^/]+).*', E'\\1')
+       END, coll_id
+    FROM r_coll_main
+   WHERE coll_name BETWEEN '/' || $1 || '/home/' || $2 AND '/' || $1 || '/home/' || $3
+      OR coll_name LIKE    '/' || $1 || '/home/' || $3 || '/%%'
+      OR coll_name BETWEEN '/' || $1 || '/trash/home/' || $2 AND '/' || $1 || '/trash/home/' || $3
+      OR coll_name LIKE    '/' || $1 || '/trash/home/' || $3 || '/%%'
+      OR coll_name BETWEEN '/' || $1 || '/trash/home/de-irods/' || $2 AND '/' || $1 || '/trash/home/de-irods/' || $3
+      OR coll_name LIKE    '/' || $1 || '/trash/home/de-irods/' || $3 || '/%%'
+      OR coll_name BETWEEN '/' || $1 || '/trash/home/ipcservices/' || $2 AND '/' || $1 || '/trash/home/ipcservices/' || $3
+      OR coll_name LIKE    '/' || $1 || '/trash/home/ipcservices/' || $3 || '/%%'
```

Comment:
These lines are what should limit the collections considered down to only those that fall within the batch, but I'd appreciate someone reviewing if I've got the right logic for that.